Nil pointer exception in probes upon re-running kubeone on bare metal cluster

### What happened?

<!-- Try to provide as much information as possible.
If you're reporting a security issue, please check the guidelines for reporting security issues:
https://github.com/kubermatic/kubeone/blob/master/CONTRIBUTING.md#reporting-a-security-vulnerability -->

I followed the instructions at https://docs.kubermatic.com/kubeone/v1.5/tutorials/creating-clusters-baremetal/ to create a bare metal cluster with kubeone. For that, I've adapted the `kubeone.yaml` example from that page.

It seems that re-running `kubeone` after the initial setup results in a nil pointer exception:

<details>

```
INFO[10:33:59 CEST] Building Kubernetes clientset...
INFO[10:33:59 CEST] Running cluster probes...
E0905 10:33:59.613224    7209 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 1 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x103caf900?, 0x104d71cc0})
	k8s.io/apimachinery@v0.24.2/pkg/util/runtime/runtime.go:75 +0x7c
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x1?})
	k8s.io/apimachinery@v0.24.2/pkg/util/runtime/runtime.go:49 +0x7c
panic({0x103caf900, 0x104d71cc0})
	runtime/panic.go:838 +0x204
k8c.io/kubeone/pkg/tasks.safeguardFlatcarMachineDeployments(0x140008bc1e0?)
	k8c.io/kubeone/pkg/tasks/probes.go:236 +0x2c
k8c.io/kubeone/pkg/tasks.safeguard(0x140008bc1e0)
	k8c.io/kubeone/pkg/tasks/probes.go:154 +0x4e0
k8c.io/kubeone/pkg/tasks.(*Task).Run.func1()
	k8c.io/kubeone/pkg/tasks/task.go:60 +0xa0
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x14000a07668, 0x1029b1bd0})
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:220 +0x28
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x103f5d020?, 0x140001aa010?}, 0x14000a076b8?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:233 +0x54
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x103c6ae01?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:226 +0x44
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x2540be400, 0x3ff6666666666666, 0x0, 0xa, 0x0}, 0x14000060000?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:421 +0x5c
k8c.io/kubeone/pkg/tasks.(*Task).Run(0x140007a52c0, 0x140008bc1e0)
	k8c.io/kubeone/pkg/tasks/task.go:55 +0x104
k8c.io/kubeone/pkg/tasks.Tasks.Run({0x140001f2700, 0x4, 0x0?}, 0x14000657958?)
	k8c.io/kubeone/pkg/tasks/tasks.go:40 +0x114
k8c.io/kubeone/pkg/cmd.runApply(0x140008bc1e0, 0x1400028ba40)
	k8c.io/kubeone/pkg/cmd/apply.go:160 +0x168
k8c.io/kubeone/pkg/cmd.applyCmd.func1(0x14000314c80?, {0x103684695?, 0x2?, 0x2?})
	k8c.io/kubeone/pkg/cmd/apply.go:93 +0xa4
github.com/spf13/cobra.(*Command).execute(0x14000314c80, {0x140007a1720, 0x2, 0x2})
	github.com/spf13/cobra@v1.4.0/command.go:856 +0x4c4
github.com/spf13/cobra.(*Command).ExecuteC(0x14000314a00)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x354
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.4.0/command.go:902
k8c.io/kubeone/pkg/cmd.Execute()
	k8c.io/kubeone/pkg/cmd/root.go:55 +0xa4
main.main()
	k8c.io/kubeone/main.go:24 +0x20
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1035b2e9c]

goroutine 1 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x1?})
	k8s.io/apimachinery@v0.24.2/pkg/util/runtime/runtime.go:56 +0xe8
panic({0x103caf900, 0x104d71cc0})
	runtime/panic.go:838 +0x204
k8c.io/kubeone/pkg/tasks.safeguardFlatcarMachineDeployments(0x140008bc1e0?)
	k8c.io/kubeone/pkg/tasks/probes.go:236 +0x2c
k8c.io/kubeone/pkg/tasks.safeguard(0x140008bc1e0)
	k8c.io/kubeone/pkg/tasks/probes.go:154 +0x4e0
k8c.io/kubeone/pkg/tasks.(*Task).Run.func1()
	k8c.io/kubeone/pkg/tasks/task.go:60 +0xa0
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x14000a07668, 0x1029b1bd0})
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:220 +0x28
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x103f5d020?, 0x140001aa010?}, 0x14000a076b8?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:233 +0x54
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x103c6ae01?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:226 +0x44
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x2540be400, 0x3ff6666666666666, 0x0, 0xa, 0x0}, 0x14000060000?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:421 +0x5c
k8c.io/kubeone/pkg/tasks.(*Task).Run(0x140007a52c0, 0x140008bc1e0)
	k8c.io/kubeone/pkg/tasks/task.go:55 +0x104
k8c.io/kubeone/pkg/tasks.Tasks.Run({0x140001f2700, 0x4, 0x0?}, 0x14000657958?)
	k8c.io/kubeone/pkg/tasks/tasks.go:40 +0x114
k8c.io/kubeone/pkg/cmd.runApply(0x140008bc1e0, 0x1400028ba40)
	k8c.io/kubeone/pkg/cmd/apply.go:160 +0x168
k8c.io/kubeone/pkg/cmd.applyCmd.func1(0x14000314c80?, {0x103684695?, 0x2?, 0x2?})
	k8c.io/kubeone/pkg/cmd/apply.go:93 +0xa4
github.com/spf13/cobra.(*Command).execute(0x14000314c80, {0x140007a1720, 0x2, 0x2})
	github.com/spf13/cobra@v1.4.0/command.go:856 +0x4c4
github.com/spf13/cobra.(*Command).ExecuteC(0x14000314a00)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x354
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.4.0/command.go:902
k8c.io/kubeone/pkg/cmd.Execute()
	k8c.io/kubeone/pkg/cmd/root.go:55 +0xa4
main.main()
	k8c.io/kubeone/main.go:24 +0x20
```
</details>

The code in question seems to be https://github.com/kubermatic/kubeone/blob/1de29e112b4abb8ea0ccf00f94f614771a6df601/pkg/tasks/probes.go#L236

It appears that this check does not account for the possibility that the `operatingSystemManager` struct is not set. Adding this to the `kubeone.yaml` file fixes the issue:

```yaml
operatingSystemManager:
  deploy: false
```

### Expected behavior

<!-- What did you expected to happen? -->

Re-running `kubeone` on a file that was successfully used to provision a cluster should not fail with a nil pointer exception.

### How to reproduce the issue?

<!-- Please provide as much information as possible, so we can reproduce the issue on our own. -->

Adapt the `kubeone.yaml` example from the documentation above as adequate and try to re-run kubeone after deploying a cluster.

### What KubeOne version are you using?

<!-- Please provide output of kubeone version here. -->

<details>

```console
$ kubeone version
{
  "kubeone": {
    "major": "1",
    "minor": "5",
    "gitVersion": "v1.5.0",
    "gitCommit": "943341700c12aab3c63371a4b19f71d608eec24a",
    "gitTreeState": "",
    "buildDate": "2022-09-05T10:33:34+02:00",
    "goVersion": "go1.18.5",
    "compiler": "gc",
    "platform": "darwin/arm64"
  },
  "machine_controller": {
    "major": "1",
    "minor": "54",
    "gitVersion": "v1.54.0",
    "gitCommit": "",
    "gitTreeState": "",
    "buildDate": "",
    "goVersion": "",
    "compiler": "",
    "platform": "linux/amd64"
  }
}
```

</details>

### Provide your KubeOneCluster manifest here (if applicable)

<!-- Providing the KubeOneCluster manifest will help us reproduce the issue.
Please make sure to redact all secrets (e.g. passwords, URLs...)! -->

Adapt example file from documentation as needed:

<details>

```yaml
apiVersion: kubeone.k8c.io/v1beta2
kind: KubeOneCluster
name: bm-cluster
versions:
  kubernetes: '1.22.5'
cloudProvider:
  none: {}

controlPlane:
  hosts:
    - publicAddress: '1.2.3.4'
      privateAddress: '172.18.0.1'
      sshUsername: root
      sshPrivateKeyFile: '/home/me/.ssh/id_rsa'
      taints:
        - key: "node-role.kubernetes.io/master"
          effect: "NoSchedule"

staticWorkers:
  hosts:
    - publicAddress: '1.2.3.5'
      privateAddress: '172.18.0.2'
      sshUsername: root
      sshPrivateKeyFile: '/home/me/.ssh/id_rsa'

# Provide the external address of your load balancer or the public addresses of
# the first control plane nodes.
apiEndpoint:
  host: ''
  port: 6443

machineController:
  deploy: false
```

</details>

### What cloud provider are you running on?

<!-- AWS, Azure, DigitalOcean, GCP, Hetzner Cloud, Nutanix, OpenStack, Equinix Metal (Packet), VMware vSphere, Other (e.g. baremetal or non-natively supported provider) -->

baremetal

### What operating system are you running in your cluster?

<!-- Ubuntu 20.04, CentOS 7, Rocky Linux 8, Flatcar Linux, ... -->

Ubuntu 20.04

### Additional information

<!-- Additional information about the bug you're reporting (optional). -->

--- a/pkg/apis/kubeone/v1beta2/defaults.go
+++ b/pkg/apis/kubeone/v1beta2/defaults.go
@@ -197,10 +197,9 @@ func SetDefaults_MachineController(obj *KubeOneCluster) {
 }
 
 func SetDefaults_OperatingSystemManager(obj *KubeOneCluster) {
-	// OSM can only be used in liaison with machine-controller
-	if obj.OperatingSystemManager == nil && (obj.MachineController != nil && obj.MachineController.Deploy) {
+	if obj.OperatingSystemManager == nil {
 		obj.OperatingSystemManager = &OperatingSystemManagerConfig{
-			Deploy: true,
+			Deploy: obj.MachineController.Deploy,
 		}
 	}
 }

--- a/pkg/apis/kubeone/validation/validation_test.go
+++ b/pkg/apis/kubeone/validation/validation_test.go
@@ -70,15 +70,15 @@ func TestValidateKubeOneCluster(t *testing.T) {
 				DynamicWorkers: []kubeoneapi.DynamicWorkerConfig{
 					{
 						Name:     "test-1",
-						Replicas: intPtr(3),
+						Replicas: pointer.Int(3),
 					},
 					{
 						Name:     "test-2",
-						Replicas: intPtr(5),
+						Replicas: pointer.Int(5),
 					},
 					{
 						Name:     "test-3",
-						Replicas: intPtr(0),
+						Replicas: pointer.Int(0),
 					},
 				},
 			},
@@ -120,15 +120,15 @@ func TestValidateKubeOneCluster(t *testing.T) {
 				DynamicWorkers: []kubeoneapi.DynamicWorkerConfig{
 					{
 						Name:     "test-1",
-						Replicas: intPtr(3),
+						Replicas: pointer.Int(3),
 					},
 					{
 						Name:     "test-2",
-						Replicas: intPtr(5),
+						Replicas: pointer.Int(5),
 					},
 					{
 						Name:     "test-3",
-						Replicas: intPtr(0),
+						Replicas: pointer.Int(0),
 					},
 				},
 			},
@@ -170,15 +170,15 @@ func TestValidateKubeOneCluster(t *testing.T) {
 				DynamicWorkers: []kubeoneapi.DynamicWorkerConfig{
 					{
 						Name:     "test-1",
-						Replicas: intPtr(3),
+						Replicas: pointer.Int(3),
 					},
 					{
 						Name:     "test-2",
-						Replicas: intPtr(5),
+						Replicas: pointer.Int(5),
 					},
 					{
 						Name:     "test-3",
-						Replicas: intPtr(0),
+						Replicas: pointer.Int(0),
 					},
 				},
 			},
@@ -220,15 +220,15 @@ func TestValidateKubeOneCluster(t *testing.T) {
 				DynamicWorkers: []kubeoneapi.DynamicWorkerConfig{
 					{
 						Name:     "test-1",
-						Replicas: intPtr(3),
+						Replicas: pointer.Int(3),
 					},
 					{
 						Name:     "test-2",
-						Replicas: intPtr(5),
+						Replicas: pointer.Int(5),
 					},
 					{
 						Name:     "test-3",
-						Replicas: intPtr(0),
+						Replicas: pointer.Int(0),
 					},
 				},
 			},
@@ -1199,15 +1199,15 @@ func TestValidateDynamicWorkerConfig(t *testing.T) {
 			dynamicWorkerConfig: []kubeoneapi.DynamicWorkerConfig{
 				{
 					Name:     "test-1",
-					Replicas: intPtr(3),
+					Replicas: pointer.Int(3),
 				},
 				{
 					Name:     "test-2",
-					Replicas: intPtr(5),
+					Replicas: pointer.Int(5),
 				},
 				{
 					Name:     "test-3",
-					Replicas: intPtr(0),
+					Replicas: pointer.Int(0),
 				},
 			},
 			expectedError: false,
@@ -1222,7 +1222,7 @@ func TestValidateDynamicWorkerConfig(t *testing.T) {
 			dynamicWorkerConfig: []kubeoneapi.DynamicWorkerConfig{
 				{
 					Name:     "test-1",
-					Replicas: intPtr(3),
+					Replicas: pointer.Int(3),
 				},
 				{
 					Name: "test-2",
@@ -1234,7 +1234,7 @@ func TestValidateDynamicWorkerConfig(t *testing.T) {
 			name: "invalid worker config (no name given)",
 			dynamicWorkerConfig: []kubeoneapi.DynamicWorkerConfig{
 				{
-					Replicas: intPtr(3),
+					Replicas: pointer.Int(3),
 				},
 			},
 			expectedError: true,
@@ -1244,7 +1244,7 @@ func TestValidateDynamicWorkerConfig(t *testing.T) {
 			dynamicWorkerConfig: []kubeoneapi.DynamicWorkerConfig{
 				{
 					Name:     "test-1",
-					Replicas: intPtr(3),
+					Replicas: pointer.Int(3),
 					Config: kubeoneapi.ProviderSpec{
 						MachineAnnotations: map[string]string{"test": "test"},
 					},
@@ -1257,7 +1257,7 @@ func TestValidateDynamicWorkerConfig(t *testing.T) {
 			dynamicWorkerConfig: []kubeoneapi.DynamicWorkerConfig{
 				{
 					Name:     "test-1",
-					Replicas: intPtr(3),
+					Replicas: pointer.Int(3),
 					Config: kubeoneapi.ProviderSpec{
 						NodeAnnotations: map[string]string{"test": "test"},
 					},
@@ -1270,7 +1270,7 @@ func TestValidateDynamicWorkerConfig(t *testing.T) {
 			dynamicWorkerConfig: []kubeoneapi.DynamicWorkerConfig{
 				{
 					Name:     "test-1",
-					Replicas: intPtr(3),
+					Replicas: pointer.Int(3),
 					Config: kubeoneapi.ProviderSpec{
 						MachineAnnotations: map[string]string{"test": "test"},
 						NodeAnnotations:    map[string]string{"test": "test"},
@@ -2209,8 +2209,4 @@ func TestValidateAssetConfiguration(t *testing.T) {
 			}
 		})
 	}
-}
-
-func intPtr(i int) *int {
-	return &i
 }

--- a/pkg/tasks/probes.go
+++ b/pkg/tasks/probes.go
@@ -23,9 +23,6 @@ import (
 	"strings"
 
 	"github.com/Masterminds/semver/v3"
-	clusterv1alpha1 "github.com/kubermatic/machine-controller/pkg/apis/cluster/v1alpha1"
-	"github.com/kubermatic/machine-controller/pkg/jsonutil"
-	providerconfigtypes "github.com/kubermatic/machine-controller/pkg/providerconfig/types"
 	"github.com/pkg/errors"
 	"gopkg.in/yaml.v2"
 
@@ -36,6 +33,10 @@ import (
 	"k8c.io/kubeone/pkg/fail"
 	"k8c.io/kubeone/pkg/kubeconfig"
 	"k8c.io/kubeone/pkg/state"
+
+	clusterv1alpha1 "github.com/kubermatic/machine-controller/pkg/apis/cluster/v1alpha1"
+	"github.com/kubermatic/machine-controller/pkg/jsonutil"
+	providerconfigtypes "github.com/kubermatic/machine-controller/pkg/providerconfig/types"
 
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"

